client/src/components/related/Outfit.jsx

```diff
@@ -2,57 +2,95 @@ import React from 'react';
 import ProductCard from './ProductCard.jsx';
 import axios from 'axios';
 
-const Outfit = (props) => {
-  const { selectedProduct, selectProduct, outfit, updateOutfit } = props;
-  const { localStorage } = window;
+class Outfit extends React.Component {
+  constructor(props) {
+    super(props);
 
-  const addToOutfit = () => {
-    if (outfit && outfit.includes(selectedProduct)) {
-      return;
+    this.addToOutfit = this.addToOutfit.bind(this);
+    this.removeFromOutfit = this.removeFromOutfit.bind(this);
+
+    this.state = {
+      outfit: []
+    };
+  }
+
+  loadOutfit() {
+    const { localStorage } = window;
+    const outfitData = localStorage.getItem('outfit');
+
+    if (outfitData) {
+      const outfit = JSON.parse(outfitData);
+      console.log('Outfit found in localStorage:', outfit);
+      this.setState({ outfit: outfit });
+    }
+  }
+
+  addToOutfit() {
+    const { localStorage } = window;
+    const { selectedProduct } = this.props;
+    const { outfit } = this.state;
+
+    if (!outfit.includes(selectedProduct.id)) {
+      console.log(`Add ${selectedProduct.id} (${selectedProduct.name}) to outfit`);
+      let newOutfit = [ selectedProduct.id, ...outfit ];
+      localStorage.setItem('outfit', JSON.stringify(newOutfit));
+      console.log('Outfit saved to localStorage:', localStorage.getItem('outfit'));
+
+      this.setState({ outfit: newOutfit });
     }
-    const newOutfit = [ selectedProduct, ...outfit ];
-    localStorage.setItem('outfit', JSON.stringify(newOutfit.map(item => item.id)));
-    console.log('Outfit saved to localStorage:', localStorage.getItem('outfit'));
-
-    updateOutfit(newOutfit);
-  };
-
-  const removeFromOutfit = (product) => {
-    const match = product.id;
-    const newOutfit = [ ...outfit ];
-    console.log('Remove', match);
-    let i = newOutfit.length;
-    while (i--) {
-      if (newOutfit[i].id === match) {
-        newOutfit.splice(i, 1);
-      }
+  }
+
+  removeFromOutfit(productId) {
+    const { localStorage } = window;
+    const { outfit } = this.state;
+    let newOutfit = [ ...outfit ];
+    const removalIndex = newOutfit.indexOf(productId);
+
+    if (removalIndex > -1) {
+      console.log(`Remove ${productId} from outfit`);
+      newOutfit.splice(removalIndex, 1);
+
+      newOutfit.length ? localStorage.setItem('outfit', JSON.stringify(newOutfit)) : localStorage.removeItem('outfit');
+      this.setState({ outfit: newOutfit });
     }
+  }
+
+  componentDidMount() {
+    this.loadOutfit();
+  }
+
+  render() {
+    const { addToOutfit, removeFromOutfit } = this;
+    const { outfit } = this.state;
+    const { selectedProduct, selectProduct, checkCache, updateCache } = this.props;
```

Comment:
Destructuring and spreading are two ES6 features that I have not quite gotten comfortable with, but seeing your usage is making me more of a fan of getting more used to using them.

Reply:
Yeah, they're super convenient for refactoring/switching between functional/class components, etc.